packages/build/src/plugins_core/deploy/buildbot_client.js

```diff
@@ -0,0 +1,86 @@
+const net = require('net')
+const { promisify } = require('util')
+
+const pEvent = require('p-event')
+
+const { addErrorInfo } = require('../../error/info')
+
+const startBuildbotClient = async function(buildbotServerSocket) {
+  if (buildbotServerSocket === undefined) {
+    return
+  }
+  try {
+    const buildbotClient = net.createConnection(buildbotServerSocket)
+    await pEvent(buildbotClient, 'connect')
+    return buildbotClient
+  } catch (error) {
+    addErrorInfo(error, { type: 'buildbotClientConnection' })
```

Comment:
Note: We will need to add this error type in our list of error types, but this can be done in a separate PR :+1: